### PR TITLE
Show all node resources in node view

### LINKED ISSUE #2762: Show ALL node resources in node view
**What would you like to be added**:
I'd like the node view to show ALL allocatable resource on the node, not just the basic CPU/mem/pod amounts.

**Why is this needed**:
In many cases kubelet reports devices, hugepages etc. as usable resources and having to dive into node yaml via edit button is, eh, bad experience. :)

Screenshot 2021-05-12 at 16 23 40

**Environment you are Lens application on:**

- Kubernetes distribution: k0s
- Desktop OS: OSX

### PR DESCRIPTION
**Pull request by vshakirova**
If I'm not mistaken, we don't always get the response in Ki

Screen Shot 2021-05-26 at 5 57 27 PM

Screen Shot 2021-05-26 at 5 59 04 PM

fixes: https://github.com/lensapp/lens/issues/2762

### CODE REVIEW

File: src/renderer/components/+nodes/node-details.tsx [hunk 1]
```diff
@@ -90,13 +98,19 @@ export class NodeDetails extends React.Component<Props> {
         <KubeObjectMeta object={node} hideFields={["labels", "annotations", "uid", "resourceVersion", "selfLink"]}/>
         <DrawerItem name="Capacity">
           CPU: {capacity.cpu},{" "}
-          Memory: {Math.floor(parseInt(capacity.memory) / 1024)}Mi,{" "}
-          Pods: {capacity.pods}
+          Memory: {this.toMi(capacity.memory)},{" "}
+          Pods: {capacity.pods},{" "}
+          Ephemeral-storage: {this.toMi(capacity["ephemeral-storage"])},{" "}
+          Hugepages-1Gi: {capacity["hugepages-1Gi"]},{" "}
+          Hugepages-2Mi: {capacity["hugepages-2Mi"]}
```
Comment (Nokel81) [inline src/renderer/components/+nodes/node-details.tsx]: Is it rather hard to read like this, could you make it into a table of sorts?

Comment (vshakirova) [reply src/renderer/components/+nodes/node-details.tsx]: Wdyt about this?
Screen Shot 2021-05-26 at 11 37 05 AM

Comment (Nokel81) [reply src/renderer/components/+nodes/node-details.tsx]: That looks amazing!

File: src/renderer/api/endpoints/nodes.api.ts [hunk 2]
```diff
@@ -56,39 +56,53 @@ export class Node extends KubeObject {
   static apiBase = "/api/v1/nodes";
 
   spec: {
-    podCIDR: string;
-    externalID: string;
+    podCIDR?: string;
+    podCIDRs?: string[];
+    providerID?: string;
+    externalID: string; // Deprecated. see: https://issues.k8s.io/61966
```
Comment (Nokel81) [inline src/renderer/api/endpoints/nodes.api.ts]: If it deprecated then it should be marked as optional and maybe even add a comment.
```suggestion
    /**
     * @deprecated see https://issues.k8s.io/61966
     */
    externalID?: string;
```

File: src/renderer/components/+nodes/node-details-resources.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+interface Props {
+  node: Node;
+  type: "allocatable" | "capacity";
+}
+
+export class NodeDetailsResources extends React.Component<Props> {
+  toMi(resource: string) {
+    if (resource.endsWith("Ki")) {
+      return `${Math.floor(parseInt(resource) / 1024)}Mi`;
```
Comment (Nokel81) [inline src/renderer/components/+nodes/node-details-resources.tsx]: Would it be better to have some decimal points?
```suggestion
      return `${(parseInt(resource) / 1024).toFixed(1)}Mi`;
```

**Review by ixrock — APPROVED** [review]

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**